stretch_core/stretch_core/stretch_driver.py

```diff
@@ -456,152 +456,159 @@ def runstop_service_callback(self, request):
         else:
             self.robot.pimu.runstop_event_reset()
 
-        return SetBoolResponse(
-            success=True,
-            message='is_runstopped: {0}'.format(request.data)
-            )
+        response.success = True,
+        response.message = 'is_runstopped: {0}'.format(request.data)
+        return response
 
-    ########### MAIN ############
+    ########### ROS Setup #######
+    def ros_setup(self):
+        self.node_name = self.get_name()
 
-    def main(self):
+        self.get_logger().info("For use with S T R E T C H (TM) RESEARCH EDITION from Hello Robot Inc.")
 
-        rospy.init_node('stretch_driver')
-        self.node_name = rospy.get_name()
-
-        rospy.loginfo("For use with S T R E T C H (TM) RESEARCH EDITION from Hello Robot Inc.")
-
-        rospy.loginfo("{0} started".format(self.node_name))
+        self.get_logger().info("{0} started".format(self.node_name))
 
         self.robot = rb.Robot()
         self.robot.startup()
 
-        mode = rospy.get_param('~mode', "position")
-        rospy.loginfo('mode = ' + str(mode))
+        self.declare_parameter('mode', "position")
+        mode = self.get_parameter('mode').value
+        self.get_logger().info('mode = ' + str(mode))
         if mode == "position":
             self.turn_on_position_mode()
         elif mode == "navigation":
             self.turn_on_navigation_mode()
 
-        self.broadcast_odom_tf = rospy.get_param('~broadcast_odom_tf', False)
-        rospy.loginfo('broadcast_odom_tf = ' + str(self.broadcast_odom_tf))
+        self.declare_parameter('broadcast_odom_tf', False)
+        self.broadcast_odom_tf = self.get_parameter('broadcast_odom_tf').value
+        self.get_logger().info('broadcast_odom_tf = ' + str(self.broadcast_odom_tf))
         if self.broadcast_odom_tf:
-            self.tf_broadcaster = tf2_ros.TransformBroadcaster()
+            self.tf_broadcaster = tf2_ros.TransformBroadcaster(self)
 
         large_ang = 45.0 * np.pi/180.0
 
-        filename = rospy.get_param('~controller_calibration_file')
-        rospy.loginfo('Loading controller calibration parameters for the head from YAML file named {0}'.format(filename))
+        self.declare_parameter('controller_calibration_file', 'not_set')
```

Comment:
Let's also convert the [stretch_driver.launch](https://github.com/hello-robot/stretch_ros2/blob/galactic/stretch_core/launch/stretch_driver.launch) launch file in this PR so that users don't have to know about and include this parameter manually. In melodic/noetic, users start stretch_driver using the launch file.